Eslint: Update dependencies and remove redundant rule setup

nikolasrieble opened the pull request:
Multiple improvements with regards to eslint configuration and setup

- Update Eslint 
- Remove deprecated https://github.com/babel/babel-eslint in favour of https://github.com/babel/babel/tree/main/eslint/babel-eslint-parser
- Add missing peer dependencies
- Remove all rules set to 'off' if no violation exists
- Remove all unused eslint-disable comments

ckifer commented:
Looks good to me. For documentations sake can we call out what deps are net new here?

Looks like
- `browserslist`
- `update-browserslist-db`
- `eslint-plugin-react-hooks`
- `babel-eslint` -> `@babel/eslint-parser`

Anything else?

nikolasrieble commented on src/util/ChartUtils.ts:
```diff
@@ -506,10 +506,10 @@ export const getDomainOfItemsWithSameAxis = (
-      Infinity,
-      -Infinity,
-    ]);
+    return domains.reduce(
+      (result, entry) => [Math.min(result[0], entry[0]), Math.max(result[1], entry[1])],
+      [Infinity, -Infinity],
+    );
```
```suggestion
    return domains.reduce((result, entry) => [Math.min(result[0], entry[0]), Math.max(result[1], entry[1])], [
      Infinity,
      -Infinity,
    ]);
```

nikolasrieble commented:
Full list: Added as dev dependencies

NEW
"eslint-plugin-react-hooks": "^4.6.0",
"update-browserslist-db": "^1.0.10",
"eslint-config-airbnb-base": "^15.0.0"
"browserslist": "^4.21.0",
CHANGES
"babel-eslint": "^10.1.0", --> "@babel/eslint-parser": "^7.19.1",

Removed from dev dependencies

ckifer commented:
>eslint-config-airbnb-base

Might be a little controversial, but I'm down to hash that out another time if Recharts already adheres to everything that isn't disabled

ckifer reviewed (approved):
LGTM

ckifer commented:
Missed that, makes sense!
